File: exercism/java/hamming/src/main/java/Hamming.java
public class Hamming {

    private String left;
    private String right;

    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length())
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        this.left = leftStrand;
        this.right = rightStrand;
    }

    int getHammingDistance() {
        int distance = 0;
        for (int i = 0; i < this.left.length(); i++)
            distance += this.left.charAt(i) == this.right.charAt(i) ? 0 : 1;
        return distance;
    }

}

File: exercism/go/hamming/hamming.go
package hamming

import "errors"

// Distance returns hamming distance between given strings
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("Input strings must be of equal length")
	}
	count := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			count++
		}
	}
	return count, nil
}
